cumulusci/core/config/base_config.py

```diff
@@ -23,7 +30,47 @@ def _load_config(self):
         """Subclasses may override this method to initialize :py:attr:`~config`"""
         pass
 
+    @classmethod
+    def allowed_names(cls) -> set:
+        properties = set(dir(cls))
+        annotations = set(getattr(cls, "__annotations__", {}))
+        return properties.union(annotations)
+
+    # long term plan is to get rid of this
     def __getattr__(self, name):
+        if not name.startswith("_"):
+            first_part = name.split("__")[0]
+            if (
+                first_part
+                not in self.all_allowed_names()
+                # and first_part not in self.config
+            ):
+                warnings.warn(
+                    f"__getattr__ on Configs is deprecated: `{first_part}` on `{self.__class__.__name__}`",
```

Comment:
This is likely to be a confusing warning, since the person seeing it is likely either a user with no awareness of how cci works internally, or a cci dev who may not understand they were using `__getattr__`.